src/platform/mbed/PlatformManagerImpl.cpp

```diff
@@ -0,0 +1,287 @@
+#include <new>
+
+#include "platform/internal/CHIPDeviceLayerInternal.h"
+
+#include <inet/InetLayer.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/PlatformManager.h>
+#include <platform/ScopedLock.h>
+#include <platform/internal/GenericPlatformManagerImpl.cpp>
+#include <rtos/ThisThread.h>
+
+#include "MbedEventTimeout.h"
+
+#if CHIP_SYSTEM_CONFIG_USE_LWIP
+#include <lwip/tcpip.h>
+#endif
+
+#define DEFAULT_MIN_SLEEP_PERIOD (60 * 60 * 24 * 30) // Month [sec]
+
+using namespace ::chip;
+using namespace ::chip::Inet;
+using namespace ::chip::System;
+
+namespace chip {
+namespace DeviceLayer {
+
+// TODO: Event and timer processing is not efficient from a memory perspective.
+// Both occupy at least 24 bytes when only 4 bytes is required.
+// An optimized designed could use a separate circular buffer to store events
+// and register a single mbed event in the event queue to process all of them.
+// A similar design can be used for timers.
+
+CHIP_ERROR PlatformManagerImpl::_InitChipStack(void)
+{
+    // Members are initialized by the stack
+    if (!mInitialized)
+    {
+        // reinitialize a new thread if it was terminated earlier
+        mLoopTask.~Thread();
+        new (&mLoopTask) rtos::Thread(osPriorityNormal, CHIP_DEVICE_CONFIG_CHIP_TASK_STACK_SIZE,
+                                      /* memory provided */ nullptr, CHIP_DEVICE_CONFIG_CHIP_TASK_NAME);
+
+        mChipTaskId = 0;
+
+        // Reinitialize the EventQueue
+        mQueue.~EventQueue();
+        new (&mQueue) events::EventQueue(event_size * CHIP_DEVICE_CONFIG_MAX_EVENT_QUEUE_SIZE);
```

Comment:
It took me minute to wrap my head around the fact this is necessary because the synchronization primitives won't have copy constructors.  This is an interesting pattern.

I appreciate the fact this pull request has been structured to allow reinitialization after shutdown.  I think that's important.

Reply:
That could probably be wrapped in a helper: 

```c++
template<typename T, typename... Args>
void reuse_object(T& v, Args&&... constructor_args)
{
    v.~T();
    new (&v) T(std::forward<Args&&>(constructor_args)...);
}
```